dataline-config/persistence/src/main/java/io/dataline/config/persistence/DefaultConfigPersistence.java

```diff
@@ -26,254 +26,133 @@
 
 import com.fasterxml.jackson.databind.JsonNode;
 import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import io.dataline.commons.enums.Enums;
 import io.dataline.commons.json.Jsons;
 import io.dataline.config.ConfigSchema;
 import java.io.IOException;
-import java.net.URI;
-import java.net.URISyntaxException;
-import java.nio.file.FileSystem;
-import java.nio.file.FileSystems;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.util.Collections;
-import java.util.HashSet;
-import java.util.Optional;
-import java.util.Set;
+import java.util.List;
 import java.util.stream.Collectors;
 import me.andrz.jackson.JsonReferenceException;
 import me.andrz.jackson.JsonReferenceProcessor;
-import org.apache.commons.io.FileUtils;
 
 // we force all interaction with disk storage to be effectively single threaded.
 public class DefaultConfigPersistence implements ConfigPersistence {
 
-  private static final String CONFIG_PATH_IN_JAR = "/json";
-  private static final String CONFIG_DIR = "schemas";
-  private static final Path configFilesRoot = getConfigFiles();
   private static final Object lock = new Object();
 
-  private final JsonSchemaValidation jsonSchemaValidation;
+  private final JsonSchemaValidator jsonSchemaValidator;
   private final Path storageRoot;
 
-  public DefaultConfigPersistence(Path storageRoot) {
+  public DefaultConfigPersistence(final Path storageRoot, final JsonSchemaValidator schemaValidator) {
```

Comment:
can we retain a constructor that just takes in the root? i'm not seeing a lot of value in having the `JsonSchemaValidator` (outside of DI for testing)

Reply:
done